native/core/src/execution/jni_api.rs

```diff
@@ -767,3 +750,20 @@ pub unsafe extern "system" fn Java_org_apache_comet_Native_traceEnd(
 ) {
     // no implementation
 }
+
+struct TraceGuard<'a> {
```

Comment:
perhaps we can have it as macros? 

```
pub fn trace_time<T, F>(label: &str, mut f: F) -> T
where
    F: FnMut() -> T,
{
    let _ = TraceGuard::new("writeSortedFileNative");
    let result = f();
    result
}
```

Reply:
Yes, I like that idea. I will try and implement this.

Reply:
I ran into some issues around lifetimes for captured variables with this approach, so perhaps it is simpler to stick with the TraceGuard.